engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneLifeCycle.mm

```diff
@@ -140,10 +145,28 @@ - (void)scene:(UIScene*)scene
     willConnectToSession:(UISceneSession*)session
            flutterEngine:(FlutterEngine*)engine
                  options:(UISceneConnectionOptions*)connectionOptions {
+  // Don't send connection options if a plugin has already used them.
+  UISceneConnectionOptions* availableOptions = connectionOptions;
+  if (self.sceneWillConnectEventHandledByPlugin) {
+    availableOptions = nil;
+  }
   BOOL handledByPlugin = [engine.sceneLifeCycleDelegate scene:scene
                                          willConnectToSession:session
-                                                      options:connectionOptions];
-  if (!handledByPlugin) {
+                                                      options:availableOptions];
+
+  // If no plugins handled this, give the application fallback a chance to handle it.
+  // Only call the fallback once since it's per application.
+  if (!handledByPlugin && !self.sceneWillConnectFallbackCalled) {
+    self.sceneWillConnectFallbackCalled = YES;
+    if ([[self applicationLifeCycleDelegate] sceneWillConnectFallback:connectionOptions]) {
+      handledByPlugin = YES;
+    }
+  }
+  if (handledByPlugin) {
+    self.sceneWillConnectEventHandledByPlugin = YES;
+  }
+
+  if (!self.sceneWillConnectEventHandledByPlugin) {
```

Comment:
Once set this flag seems to never resets back to NO. Just want to make sure that we have the guarantee that each `FlutterPluginSceneLifeCycleDelegate` can only be associated with one scene and thus will never receive a second `willConnect` event?

Reply:
That is my understanding, yeah. If the scene is destroyed, for example, a new `FlutterSceneDelegate` and therefore `FlutterPluginSceneLifeCycleDelegate` is created and will receive a new `willConnect` event